### ISSUE TITLE
Kinect image and delay

### ISSUE THREAD

**Issue by alduxvm** [nxdefiant/ros_image_to_rtsp#4]
Hi!

I'm testing your node using a kinect v1 (with the same topic) and I'm getting an error on the console, which is:
```[ INFO] [1509994047.569972956]: New RTSP client
[ INFO] [1509994047.571572958]: Client connected: /rgb
x264 [error]: baseline profile doesn't support 4:4:4
```
I can still the stream, an example of what I see when I open the stream using vlc is here:
<img width="692" alt="screen shot 2017-11-06 at 1 47 34 pm" src="https://user-images.githubusercontent.com/8823379/32458126-55b64fea-c2f9-11e7-80d2-1da7cf7a51fd.png">

I'm using ubuntu 16.04 on an odroid xu4 and I installed gst1.0 as in your instructions: `sudo apt-get install libgstrtspserver-1.0-dev gstreamer1.0-plugins-good gstreamer1.0-plugins-bad gstreamer1.0-plugins-ugly` 

Also, I can see a delay of 10 to 20 seconds on the image... maybe gst is not properly installed? or what can the issue be? 

Thanks a lot!

**Comment by nxdefiant** [comment]
The error shouldn't be the problem. This looks like a resolution or color format mismatch. Can you add the output of

> rostopic echo -n 1 --noarr /camera/rgb/image_raw

Is the delay increasing? If so you might want to check the cpu usage as it might be too much for the cpu.

**Comment by nxdefiant** [comment]
Also when encoding on an ARM controller I recommend to use a hardware encoder instead of the cpu (which is why this node exists in the first place), for odroid the keyword [seems to be](https://forum.odroid.com/viewtopic.php?f=77&t=5960) [v4l2video8enc](https://forum.odroid.com/viewtopic.php?f=77&t=5960&start=50)

**Comment by alduxvm** [comment]
For the `rostopic echo -n 1 --noarr /camera/rgb/image_raw`:

```header: 
  seq: 0
  stamp: 
    secs: 1510000564
    nsecs: 293323078
  frame_id: camera_rgb_optical_frame
height: 480
width: 640
encoding: bayer_grbg8
is_bigendian: 0
step: 640
data: <array type: uint8, length: 307200>
---
```

I'm going to check the hardware controller just now... will let you know.

**Comment by nxdefiant** [comment]
Yeah, the launch file is configured for a 320x240 image while your  /camera/rgb/image_raw topic is 640x480. Thats why the the image is disturbed. Two options:

- Either change the value for width and height in the caps entry of the launch file
- or you could try the new commit 665cf56 which reads the caps from the ros topic.

**Comment by alduxvm** [comment]
I updated the repository, did the compilation again, installed the .debs from the odroid link, and now I have this:
<img width="639" alt="screen shot 2017-11-06 at 3 49 58 pm" src="https://user-images.githubusercontent.com/8823379/32463227-2a37de4a-c30a-11e7-9449-81853ae62a10.png">
and in the console, still the same error:
```[ INFO] [1510001244.515321415]: New RTSP client
[ INFO] [1510001244.516869170]: Client connected: /rgb
x264 [error]: baseline profile doesn't support 4:4:4
```

**Comment by nxdefiant** [comment]
Please only one change at a time. Unfortunately I don't have a ros enabled odroid atm, so lets try without the custom odroid packages first. If it still does not work please add a 5-10s bag file of the /camera/rgb/image_raw  topic.
